specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json

```diff
@@ -402,8 +402,127 @@
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SignalRService/SignalR/{resourceName}/restart": {
+      "post": {
+        "tags": [
+          "SignalR"
+        ],
+        "description": "Operation to restart a SignalR service.",
+        "operationId": "SignalR_Restart",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/SignalRServiceName"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted. The response indicates the restart operation is performed in the background."
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "SignalR_Restart": {
+            "$ref": "./examples/SignalR_Restart.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SignalRService/SignalR/{resourceName}/switchFeatures": {
+      "post": {
+        "tags": [
+          "SignalR"
+        ],
+        "description": "Switch on/off SignalR resource features.",
+        "operationId": "SignalR_SwitchFeatures",
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "Parameters that describes the SignalR resource features opreation.",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/SignalRFeaturesParameters"
+            }
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/SignalRServiceName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The response indicates the operation is successful and contains a list of features.",
+            "schema": {
+              "$ref": "#/definitions/SignalRFeatureList"
+            }
+          },
+          "202": {
+            "description": "Accepted. The response indicates a longer time operation is performed in the background."
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "SignalR_SwitchFeatures": {
+            "$ref": "./examples/SignalR_SwitchFeatures.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SignalRService/SignalR/{resourceName}/listFeatures": {

```

Comment:
make it features instead of listFeatures. Also, I want to understand more about this resource type? What are these features? Can you give an example? From what i think it does, it might make sense to have these "features" as properties on the resource itself which can be enabled/disabled by PUT/PATCH operations on the signalR resource itself.

Reply:
Basically a SignalR resource has some backend containers running in ACS, where each container is running a SignalR runtime written in C#. Those features are actually configurations of the SignalR runtime. For now, we only have one feature named `ServiceMode`(either Default, Classic or Serverless), which corresponds to different customer scenarios. SignalR runtime will read feature values and behave correspondingly. 

For us, SignalR service management and Service runtime configuration is quite separated. As I said, the runtime is written in C#(and SignalR is essentially a asp.net library) which means we actually have many C#/SignalR configurations that can change the realtime behavior. We may gradually open some settings to customers. That's why we may want a new resource type dedicated for these kind of things, for better separation and flexibility

Reply:
After some internal discussion, we decided to deprecate those two APIs: `ListFeature` and `SwitchFeature`. We will use propertes instead, just like you said. So I removed those two spec from swagger file